lib/parser/lexer/literal.rb

```diff
@@ -183,11 +183,20 @@ def end_interp_brace_and_try_closing
       (@interp_braces == 0)
     end
 
-    def extend_string(string, ts, te)
-      @buffer_s ||= ts
-      @buffer_e = te
+    if RUBY_ENGINE == 'opal'
```

Comment:
I strongly oppose having special implementations for different Ruby engines.

Reply:
Opal is not a Ruby engine. 😃 It's a Ruby to JS compiler with quite a few differences from "standard" Ruby.